File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_5_dip/after/Midfielder.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_5_dip.after;

/**
 * A Low-Level Detail.
 * * ARCHITECTURE NOTE: Just like the Forward, this concrete class depends 
 * on the Player abstraction rather than being hardcoded into the Coach.
 */
public class Midfielder implements Player {
    public void performAction() {
        System.out.println("  [Action] Midfielder is controlling the game.");
    }
}

File: Chapter 03/C#/Section_3_3_5_DIP/After/Midfielder.cs
using System;

namespace Chapter03.DIP.After;
/// <summary>
/// A Low-Level Detail.
/// 
/// ARCHITECTURE NOTE: The concrete implementation now depends on the abstraction 
/// (IPlayer). The flow of dependency has been inverted.
/// </summary>
public class Midfielder : IPlayer
{
    public void PerformAction() => Console.WriteLine("  [Action] Midfielder is controlling the game.");
}